nvtabular/io/csv.py

```diff
@@ -36,8 +36,23 @@ def __init__(self, paths, part_size, storage_options=None, **kwargs):
         if len(self.paths) == 1 and self.fs.isdir(self.paths[0]):
             self.paths = self.fs.glob(self.fs.sep.join([self.paths[0], "*"]))
 
-    def to_ddf(self, columns=None, cpu=False):
-        _lib = dd if cpu else dask_cudf
-        if columns:
-            return _lib.read_csv(self.paths, chunksize=self.part_size, **self.csv_kwargs)[columns]
-        return _lib.read_csv(self.paths, chunksize=self.part_size, **self.csv_kwargs)
+    def to_ddf(self, columns=None, cpu=None):
+
+        # Check if we are using cpu
+        cpu = self.cpu if cpu is None else cpu
+        if cpu:
+            if columns:
+                return dd.read_csv(self.paths, blocksize=self.part_size, **self.csv_kwargs)[columns]
+            return dd.read_csv(self.paths, blocksize=self.part_size, **self.csv_kwargs)
+        else:
+            if columns:
+                return dask_cudf.read_csv(self.paths, chunksize=self.part_size, **self.csv_kwargs)[
+                    columns
+                ]
+            return dask_cudf.read_csv(self.paths, chunksize=self.part_size, **self.csv_kwargs)
```

Comment:
super minor nitpick: I think it'd be cleaner to minimize the number of branches here - maybe something like this instead?

```suggestion
        if cpu:
            ddf = dd.read_csv(self.paths, blocksize=self.part_size, **self.csv_kwargs)
        else:
            ddf = dask_cudf.read_csv(self.paths, chunksize=self.part_size, **self.csv_kwargs)
        if columns:
            ddf = ddf[columns]
        return ddf
```